packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -293,6 +293,11 @@ export abstract class BucketBase extends cdk.Construct implements IBucket {
    */
   protected abstract autoCreatePolicy = false;
 
+  /**
+   * The block public access configuration of this bucket.
+   */
+  protected abstract blockPublicAccess?: BlockPublicAccess;
```

Comment:
I actually think this should have remained `disallowPublicAccess` because it reflects the intent to disallow `grantPublicAccess` and not the specific type of public access blocked during configuration. Otherwise, you go back to faking it with the imported resource...